src/numerics/petsc_matrix.C

```diff
@@ -80,6 +80,7 @@ namespace libMesh
 template <typename T>
 PetscMatrix<T>::PetscMatrix(const Parallel::Communicator & comm_in) :
   SparseMatrix<T>(comm_in),
+  _mat(NULL),
```

Comment:
I'm a bit surprised we didn't run into issues with not initializing `_mat` in the past. Definitely seems like the right thing to do, but we should probably use `nullptr` for consistency with C++11.

Reply:
I'm not too surprised we haven't run into errors; we generally init() pretty quickly after construction.

As for warnings?  IIRC this isn't caught by -Wuninitialized.  I think it is caught by -Weffc++, but so are so many false positives (even in libstdc++ headers, according to `man gcc`) that we never use that.

Reply:
In pretty much any typical PETSc code you would see code like:
```
Mat mat;
```
i.e. without explicit nullptr-initialization, so we were at least mimicking that behavior previously.